modules/json/src/parsers/convert-json-expression.js

```diff
@@ -0,0 +1,74 @@
+import {get} from '../utils/get';
+
+// expression-eval: Small jsep based expression parser that supports array and object indexing
+import expressionEval from 'expression-eval';
+
+class EvaluationError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = 'EvaluationError';
+  }
+}
+
+const SINGLE_IDENTIFIER_REGEX = /^\s*[A-Za-z]+\s*$/;
```

Comment:
No underscores or numbers?  Should e.g. `ID_0` match?

Add a comment about what identifier syntax we are supporting? JavaScript?

Reply:
As I read a little more into this, the rules are a bit more complicated than this regex allows (all of unicode is valid as a Javascript identifier and numbers are not valid as the first character of an identifier, for example). Would we be better off letting expression-eval parse the identifier and tell us whether it's valid or not? Any issues you can foresee with that?

Reply:
Actually, never mind–you propose this below